osh/hub/osh_xmlrpc/worker.py

```diff
@@ -76,6 +79,28 @@ def finish_task(request, task_id):
                 task.fail_task()
 
 
+@validate_worker
+def open_task(request, task_id):
+    response = kobo_open_task(request, task_id)
+
+    if settings.ENABLE_SINGLE_USE_WORKERS:
+        task = Task.objects.get(id=task_id)
+
+        Task.create_shutdown_task("worker/" + task.worker.name, task.worker.name, kill=False)
```

Comment:
@lzaoral This method may worker better, but would spam the tasks view with `ShutdownWorker` tasks. Shall we modify the view to skip such tasks?

Reply:
That view is provided by `kobo` and filtering on the `method` task attribute is not implemented.